internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepository.java

```diff
@@ -128,11 +132,71 @@ private Object doSchemaOperation(String storeName, Function<SchemaData, Object>
     }
   }
 
-  private void mayRegisterAndPopulateMetadataSchema(Store store, SchemaData schemaData) {
+  void maybeRegisterAndPopulateRmdSchema(Store store, SchemaData schemaData) {
     if (store.isActiveActiveReplicationEnabled()) {
       String storeName = store.getName();
-      accessor.subscribeReplicationMetadataSchemaCreationChange(storeName, replicationMetadataSchemaChildListener);
-      accessor.getAllReplicationMetadataSchemas(storeName).forEach(schemaData::addReplicationMetadataSchema);
+      getAccessor().subscribeReplicationMetadataSchemaCreationChange(storeName, replicationMetadataSchemaChildListener);
+      getAccessor().getAllReplicationMetadataSchemas(storeName).forEach(schemaData::addReplicationMetadataSchema);
+    }
+  }
+
+  void maybeRegisterAndPopulateUpdateSchema(Store store, SchemaData schemaData) {
+    if (store.isWriteComputationEnabled()) {
+      String storeName = store.getName();
+      getAccessor().subscribeDerivedSchemaCreationChange(storeName, derivedSchemaChildListener);
+      getAccessor().getAllDerivedSchemas(storeName).forEach(schemaData::addDerivedSchema);
+    }
+  }
+
+  SchemaEntry forceRefreshSupersetSchemaWithRetry(String storeName) {
+    Store store = getStoreRepository().getStore(storeName);
+    int supersetSchemaId = store.getLatestSuperSetValueSchemaId();
+    getSchemaLock().writeLock().lock();
+    SchemaData schemaData = getSchemaMap().get(getZkStoreName(storeName));
```

Comment:
Again check here is schemaData already contains the superset schema id or not.

Reply:
It is not achievable here, since we move the lock into retry block. Also, this is not necessary as we checked this before.